### PR TITLE
[eclipse/xtext-core#1059] updated to lsp4j 0.7.0

### PR DESCRIPTION
**Pull request by cdietrich**
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend [hunk 1]
```diff
@@ -48,7 +48,11 @@ class DocumentSymbolMapper {
 	 */
 	def DocumentSymbol toDocumentSymbol(EObject object) {
 		return new DocumentSymbol => [
-			name = nameProvider.getName(object);
+			val pName = nameProvider.getName(object) {
+				if (pName !== null) {
+					name = pName;
+				}
+			}
```
Comment (spoenemann) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: This code looks strange. What are the curly braces around the `if` statement doing?

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: arrrg. will correct

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: corrected

Comment (kittaakos) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: @cdietrich, can you please explain why do we need the `null` check here?

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: cause of this fancy new check:
https://github.com/eclipse/lsp4j/blob/58bd73af0b7b62610e626aaaf1a625c1d57ccb1d/org.eclipse.lsp4j/src/main/xtend-gen/org/eclipse/lsp4j/DocumentSymbol.java#L107

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: will add a check for the others too

Comment (kittaakos) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: I see. :+1: Thank you! Quite an inconvenient new lsp4j feature.

Comment (kittaakos) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: Then we need the `null` check for almost all other setters, don't we? Why do we have this new fail fast "feature" in lsp4j, @spoenemann?

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: done

Comment (kittaakos) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: Well, I think this is an LSP4J issue, we use it in JDT too, perhaps other projects as well, we cannot just change the runtime behavior from `0.6.0` to `0.7.0`.

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: yes and no. the old impl was to use the stuff created here as "schmierzettel" and sort out the bad ones later. see org.eclipse.xtext.ide.server.symbol.HierarchicalDocumentSymbolService.isValid(DocumentSymbol)

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: the same is still used with my change. it avoid the illegalarguments only

Comment (spoenemann) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: Here's the origin issue: https://github.com/eclipse/lsp4j/issues/284

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: @kittaakos how to proceed? go with how it is now and open a lsp4j issue?

Comment (kittaakos) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: I would either revert the behavior change in LSP4J or accept it and add the `null` checks to the client code, but opening a new LSP4J ticket does not make much sense to me. What do you think, @spoenemann? Is it urgent to decide on this, @cdietrich?

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: well i dont know when and if people might want to update lsp4j with xtext

Comment (spoenemann) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: But do we really need all those null guards? Have you confirmed that `nameProvider`, `kindProvider` and `rangeProvider` can actually return null values? If yes, isn't that a bug in those providers? In those cases we could as well just let the exception bubble up so an error response message is sent to the client.

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: @spoenemann i added null guards to the places that currently can return null

Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/symbol/DocumentSymbolMapper.xtend]: an as i said: the current impl uses schmiezettel logik and throws out the improper objects later

**Pull request merged**